src/Store.js

```diff
@@ -149,4 +197,27 @@ export default class Store {
       this.findOne({ [identifier]: itemKey[identifier] }).remove(source)));
   }
 
+  _workFetchResult(storeItems, fetchItems, count, source) {
+    // this is not very performant right now. But as we want to release the first
+    // beta and aren't sure if this behavior stays the same anyway we go with this for now.
+    const identifier = this.schema.getKeyIdentifierFor(source.AS_TARGET);
+    const itemsToRemove = [];
+    storeItems.forEach((storeItem) => {
+      if (storeItem[identifier]) {
+        const fetchItem = fetchItems.find(item => storeItem[identifier] === item[identifier]);
+        // this is kinda bad style but as the array is never used after this
+        // i think we should be fine
+        if (fetchItem) {
+          fetchItems.splice(fetchItems.indexOf(fetchItem), 1);
```

Comment:
Also a more performant version of this would be to save all handled items in a separate array. Then after the loop go through the whole array and "subtract" these. Because the "handled items" array should be a subset with the same order as the fetchItems array this can be done in O(n).